File: todo/src/main/java/com/king/todo/inject/component/AppComponent.java
package com.king.todo.inject.component;

import android.content.Context;

import com.king.todo.inject.data.api.Api;
import com.king.todo.inject.data.sql.DatabaseApi;
import com.king.todo.inject.module.AppModule;
import com.king.todo.inject.module.DataModule;
import com.king.todo.inject.module.NetWorkModule;
import com.king.todo.inject.qualifier.context.AppContext;
import com.king.todo.inject.scope.ApplicationScope;
import com.king.todo.ui.MoneyApplication;

import dagger.Component;

/**
 * project：cutv_ningbo
 * description：
 * create developer： admin
 * create time：10:08
 * modify developer：  admin
 * modify time：10:08
 * modify remark：
 *
 * @version 2.0
 */

@ApplicationScope
@Component(modules={AppModule.class, NetWorkModule.class,DataModule.class})
public interface AppComponent {
    @AppContext Context context();
    Api getApi();
    DatabaseApi getDatabaseApi();
    void inject(MoneyApplication application);
}

File: todokotlin/src/main/java/com/union/bangbang/todokotlin/dagger/component/AppComponent.kt

import android.app.Application
import android.content.Context
import com.union.bangbang.todokotlin.TodoApplication
import com.union.bangbang.todokotlin.dagger.module.*
import dagger.BindsInstance
import dagger.Component
import dagger.Provides
import dagger.android.AndroidInjectionModule
import javax.inject.Named
import javax.inject.Qualifier
import javax.inject.Singleton

@Singleton
@Component(modules = [
    (AppModule::class),
    (AndroidInjectionModule::class),
    (DataModule::class),
    (ActivityModule::class),
    (FragmentModule::class),
    (ViewModelModule::class)])
interface AppComponent {

    @Component.Builder
    interface Builder {
        @BindsInstance
        fun application(application: Application): Builder
//        @BindsInstance
//        fun appModule(appModule: AppModule):Builder
        fun build(): AppComponent
    }
    fun getContext(): Context

    fun inject(application: TodoApplication)
}